editor/src/components/custom-code/code-file.ts

```diff
@@ -254,3 +273,169 @@ export function codeCacheToBuildResult(cache: {
+  normalisedPath: StaticInstancePath
+  filePath: string
+  textFile: TextFile
+}
+
+export function normalisePathSuccess(
+  normalisedPath: StaticInstancePath,
+  filePath: string,
+  textFile: TextFile,
+): NormalisePathSuccess {
+  return {
+    type: 'NORMALISE_PATH_SUCCESS',
+    normalisedPath: normalisedPath,
+    filePath: filePath,
+    textFile: textFile,
+  }
+}
+
+export interface NormalisePathEndsAtDependency {
+  type: 'NORMALISE_PATH_ENDS_AT_DEPENDENCY'
+  dependency: string
+}
+
+export function normalisePathEndsAtDependency(dependency: string): NormalisePathEndsAtDependency {
+  return {
+    type: 'NORMALISE_PATH_ENDS_AT_DEPENDENCY',
+    dependency: dependency,
+  }
+}
+
+export interface NormalisePathError {
+  type: 'NORMALISE_PATH_ERROR'
+  errorMessage: string
+}
+
+export function normalisePathError(errorMessage: string): NormalisePathError {
+  return {
+    type: 'NORMALISE_PATH_ERROR',
+    errorMessage: errorMessage,
+  }
+}
+
+export interface NormalisePathUnableToProceed {
+  type: 'NORMALISE_PATH_UNABLE_TO_PROCEED'
+}
+
+export function normalisePathUnableToProceed(): NormalisePathUnableToProceed {
+  return {
+    type: 'NORMALISE_PATH_UNABLE_TO_PROCEED',
+  }
+}
+
+export interface NormalisePathImportNotFound {
+  type: 'NORMALISE_PATH_IMPORT_NOT_FOUND'
+}
+
+export function normalisePathImportNotFound(): NormalisePathImportNotFound {
+  return {
+    type: 'NORMALISE_PATH_IMPORT_NOT_FOUND',
+  }
+}
+
+export type NormalisePathResult =
+  | NormalisePathError
+  | NormalisePathUnableToProceed
+  | NormalisePathImportNotFound
+  | NormalisePathEndsAtDependency
+  | NormalisePathSuccess
+
+export function normalisePathToUnderlyingTarget(
+  projectContents: ProjectContentTreeRoot,
+  currentFilePath: string,
+  elementPath: InstancePath,
+): NormalisePathResult {
+  const currentFile = getContentsTreeFileFromString(projectContents, currentFilePath)
+  if (isTextFile(currentFile)) {
+    if (
+      currentFile.fileContents.revisionsState === RevisionsState.CodeAhead ||
+      !isParseSuccess(currentFile.fileContents.parsed)
+    ) {
+      // As the code is ahead this would potentially be looking at a path
+      // which now doesn't exist.
+      return normalisePathUnableToProceed()
+    } else {
+      const staticPath = TP.dynamicPathToStaticPath(elementPath)
+      const potentiallyDroppedFirstSceneElementResult = TP.dropFirstScenePathElement(staticPath)
+      if (potentiallyDroppedFirstSceneElementResult.droppedScenePathElements == null) {
+        // As the scene path is empty, there's no more traversing to do, the target is in this file.
+        return normalisePathSuccess(staticPath, currentFilePath, currentFile)
+      } else {
+        const droppedPathPart = potentiallyDroppedFirstSceneElementResult.droppedScenePathElements
+        if (droppedPathPart.length === 0) {
+          return normalisePathError(
+            `Unable to handle empty scene path part for ${TP.toString(elementPath)}`,
```

Comment:
Is this the difference between `scenePath([])` and `scenePath([[]])`?

Reply:
Yes.